src/Mock.ts

```diff
@@ -26,26 +28,23 @@ export const JestMockExtended = {
     },
 };
 
-export interface CalledWithMock<T, Y extends any[]> extends jest.Mock<T, Y> {
-    calledWith: (...args: Y | MatchersOrLiterals<Y>) => jest.Mock<T, Y>;
+export interface CalledWithMock<T extends FunctionLike> extends jest.Mock<T> {
+    calledWith: (...args: [...Parameters<T>] | MatchersOrLiterals<[...Parameters<T>]>) => jest.Mock<T>;
```

Comment:
I think this change here is why the 2 // @ts-expect-error are required in Mock.spec.ts, and I suspect that is a breaking change. 

Just seeing if I can improve this to avoid that.